internal/engine/execution/global.go

```diff
@@ -36,6 +38,19 @@ type GlobalContext struct {
 
 var ErrDatasetNotFound = fmt.Errorf("dataset not found")
 
+func InitializeEngine(ctx context.Context, tx sql.DB) error {
+	upgradeFns := map[int64]versioning.UpgradeFunc{
+		-1: initTables,
```

Comment:
Similar to my comments on account store, this seems like a version 0 to me. -1 is meant as a fast-path for new nodes that have never initialized, and need to initialize